packages/botfuel-dialog/src/nlus/intent.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2017 - present, Botfuel (https://www.botfuel.io).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const logger = require('logtown')('Intent');
+const SdkError = require('../errors/sdk-error');
+
+/** Intent class */
+class Intent {
+  /**
```

Comment:
I Would have define Intent types as static properties (as in the Dialog class) and I would used it in the other classes where this is needed:

```javascript
static TYPE_QNA = 'QnA';
static TYPE_INTENT = 'Intent';

get TYPE_QNA() {
  return Intent.TYPE_QNA;
}

get TYPE_INTENT() {
  return Intent.TYPE_INTENT;
}
```

By doing this you make sure that the value of these types are defined in only one place which is very useful if the value have to change in the future.

Reply:
Thank you. Good point. Will apply this.

Reply:
thank you. good point!